src/v2_0/rw/elog.cpp

```diff
@@ -84,54 +84,82 @@ namespace abb ::rws ::v2_0 ::rw ::elog
+        bool separator_flag = false;
+        for (auto node : nodes)
+        {
+            if (separator_flag)
+                ss << separator;
+            separator_flag = true;
+            ss << node->innerText();
+        }
+        return ss.str();
+    }
+
+    ElogMessage parseElogNode(Poco::XML::Node* node)
     {
-        Poco::AutoPtr<Poco::XML::Document> rws_result = parseXml(xml_string);
         ElogMessage message;
-        message.messageType = makeElogMessageType(std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "msgtype"))));
-        message.code = std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "code")));
-        message.sourceName;
-        std::istringstream ss(xmlFindTextContent(rws_result, XMLAttribute("class", "tstamp")));
+
+        std::string title = xmlNodeGetAttributeValue(node, "title");
+        int sep_pos = title.find_last_of("/");
+        message.domain = std::stoi(title.substr(9, sep_pos - 9));
+        message.sequenceNumber = std::stoi(title.substr(sep_pos + 1));
+        message.messageType = makeElogMessageType(std::stoi(xmlFindTextContent(node, XMLAttribute("class", "msgtype"))));
+        message.code = std::stoi(xmlFindTextContent(node, XMLAttribute("class", "code")));
+        message.sourceName = xmlFindTextContent(node, XMLAttribute("class", "src-name"));
+        std::istringstream ss(xmlFindTextContent(node, XMLAttribute("class", "tstamp")));
         ss >> std::get_time(&message.timestamp, "%Y-%m-%d T %H:%M:%S");
-        message.title = xmlFindTextContent(rws_result, XMLAttribute("class", "title"));
-        message.desc = xmlFindTextContent(rws_result, XMLAttribute("class", "desc"));
-        message.conseqs = xmlFindTextContent(rws_result, XMLAttribute("class", "conseqs"));
-        message.causes = xmlFindTextContent(rws_result, XMLAttribute("class", "causes"));
-        message.actions = xmlFindTextContent(rws_result, XMLAttribute("class", "actions"));
+        message.title = concatTextFromNodes(xmlFindNodes(node, XMLAttribute("class", "title")));
+        message.desc = concatTextFromNodes(xmlFindNodes(node, XMLAttribute("class", "desc")));
+        message.conseqs = concatTextFromNodes(xmlFindNodes(node, XMLAttribute("class", "conseqs")));
+        message.causes = concatTextFromNodes(xmlFindNodes(node, XMLAttribute("class", "causes")));
+        message.actions = concatTextFromNodes(xmlFindNodes(node, XMLAttribute("class", "actions")));
 
-        int argc = std::stoi(xmlFindTextContent(rws_result, XMLAttribute("class", "argc")));
+        int argc = std::stoi(xmlFindTextContent(node, XMLAttribute("class", "argc")));
 
         for (int i = 1; i <= argc; i++)
         {
             std::stringstream arg_name;
             arg_name << "arg" << i;
-            auto node = xmlFindNodes(rws_result, XMLAttribute("class", arg_name.str()))[0];
+            auto arg_node = xmlFindNodes(node, XMLAttribute("class", arg_name.str()))[0];
```

Comment:
Here if no nodes are found we end up with SIGSEGV, maybe we should throw a more readable exception?